knotx-knots/knotx-knot-service/src/main/java/com/cognifide/knotx/knot/service/service/ServiceEngine.java

```diff
@@ -55,14 +60,19 @@ public ServiceEngine(EventBus eventBus, ServiceKnotConfiguration serviceConfigur
   }
 
   public ServiceEntry mergeWithConfiguration(final ServiceEntry serviceEntry) {
-    return configuration.getServices().stream()
-        .filter(service -> serviceEntry.getName().matches(service.getName()))
-        .findFirst().map(metadata ->
-            serviceEntry.setAddress(metadata.getAddress())
-                .mergeParams(metadata.getParams())
-                .overrideCacheKey(metadata.getCacheKey())
-        )
-        .get();
+    Optional<ServiceKnotConfiguration.ServiceMetadata> serviceMetadata = configuration.getServices().stream()
+            .filter(service -> serviceEntry.getName().matches(service.getName()))
+            .findFirst();
+    if (serviceMetadata.isPresent()) {
+      return serviceMetadata.map(metadata ->
+              serviceEntry.setAddress(metadata.getAddress())
+                      .mergeParams(metadata.getParams())
+                      .overrideCacheKey(metadata.getCacheKey())
+      ).get();
```

Comment:
return serviceMetadata.map(metadata ->
              serviceEntry.setAddress(metadata.getAddress())
                      .mergeParams(metadata.getParams())
                      .overrideCacheKey(metadata.getCacheKey())
      ).get();

The lamba expression in this return statement appears to mutate an object passed as a parameter. I find this a little difficult to follow. Perhaps you meant to build a new object of the same type?